pkg/apis/certificates/validation/validation.go

```diff
@@ -152,6 +184,22 @@ func validateCertificateSigningRequest(csr *certificates.CertificateSigningReque
 	if len(csr.Spec.Usages) == 0 {
 		allErrs = append(allErrs, field.Required(specPath.Child("usages"), "usages must be provided"))
 	}
+	if !opts.allowUnknownUsages {
+		for i, usage := range csr.Spec.Usages {
+			if !allValidUsages.Has(string(usage)) {
+				allErrs = append(allErrs, field.NotSupported(specPath.Child("usages").Index(i), usage, allValidUsages.List()))
+			}
+		}
+	}
+	if !opts.allowDuplicateUsages {
+		seen := map[certificates.KeyUsage]bool{}
```

Comment:
It's weird that there is a function that checks this and this inlines almost identical logic.  Not so weird I want you to change, but still weird.

Reply:
I thought briefly about refactoring, but the early exit and lack of care about the index made it awkward to combine